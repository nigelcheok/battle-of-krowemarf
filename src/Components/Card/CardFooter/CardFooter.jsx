import React from 'react';

import { GroupButton } from '../../Button/GroupButton/GroupButton';

export function CardFooter(props) {
  return (
    <div className="card-footer" style={CardFooterStyle()}>
      { props.clanDetails &&
        <div>
          <GroupButton label="Stars" value={props.clanDetails.starsCount} icon="fa-star"/>
          <GroupButton label="Issues" value={props.clanDetails.issuesCount} icon="fa-exclamation-circle"/>
          <GroupButton label="Forks" value={props.clanDetails.forksCount} icon="fa-code-branch"/>
          <GroupButton label="Pull Requests" value={props.clanDetails.pullRequestsCount}
                       icon="fa-code-branch vertical-flip" flipIcon/>
        </div>
      }
    </div>
  );
}


function CardFooterStyle() {
  return {
    borderTop: '0',
  }
}
