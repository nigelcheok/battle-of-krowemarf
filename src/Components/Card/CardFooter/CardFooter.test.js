import React from 'react';
import { shallow } from 'enzyme';
import { CardFooter } from './CardFooter';

describe('CardFooter', () => {
  it('should render correctly with no props', () => {
    const component = shallow(<CardFooter/>);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with props clanDetails', () => {
    const clanDetailsData = {
      clan: undefined,
      id: undefined,
      name: undefined,
      starsCount: 1,
      forksCount: 2,
      issuesCount: 3,
      pullRequestsCount: 4,
    };

    const component = shallow(<CardFooter clanDetails={clanDetailsData}/>);
    expect(component).toMatchSnapshot();
  });
});