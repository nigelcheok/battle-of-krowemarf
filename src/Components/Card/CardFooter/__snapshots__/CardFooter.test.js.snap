// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooter should render correctly with no props 1`] = `
<div
  className="card-footer"
  style={Object {}}
/>
`;

exports[`CardFooter should render correctly with props clanDetails 1`] = `
<div
  className="card-footer"
  style={Object {}}
>
  <div>
    <GroupButton
      icon="fa-star"
      label="Stars"
      value={1}
    />
    <div
      style={
        Object {
          "display": "inline-block",
          "padding": "0 20px",
        }
      }
    />
    <GroupButton
      icon="fa-exclamation-circle"
      label="Issues"
      value={3}
    />
    <div
      style={
        Object {
          "display": "inline-block",
          "padding": "0 20px",
        }
      }
    />
    <GroupButton
      icon="fa-code-branch"
      label="Forks"
      value={2}
    />
    <div
      style={
        Object {
          "display": "inline-block",
          "padding": "0 20px",
        }
      }
    />
    <GroupButton
      flipIcon={true}
      icon="fa-code-branch vertical-flip"
      label="Pull Requests"
      value={4}
    />
  </div>
</div>
`;
